src/resolve-template.js

```diff
@@ -0,0 +1,90 @@
+var path = require("path");
+var fileSystem = require('fs');
+var readdir = require('recursive-readdir');
+var ContextElementDependency = require('webpack/lib/dependencies/ContextElementDependency');
+var assign = Object.assign || require('object.assign');
+
+var cheerio = require('cheerio');
+
+/**
+ * Generates key-value pairs or <require from="paths"> and their corresponding files.
+ */
```

Comment:
Those are the ways that are defined by the core. It's possible for developers to invent new ones ;)

Once this is built, it might also be a good idea to extract the core logic so its independent from webpack, as @atsu85 says, there are a lot of other things we could do with this probably. We don't need to worry about that just yet, but lets keep it in mind for the future.